RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc

```diff
@@ -10,13 +10,26 @@
 
 using namespace ticl;
 
-PatternRecognitionbyCA::PatternRecognitionbyCA(const edm::ParameterSet &conf) : PatternRecognitionAlgoBase(conf) {
+PatternRecognitionbyCA::PatternRecognitionbyCA(const edm::ParameterSet &conf, const CacheBase *cache)
+    : PatternRecognitionAlgoBase(conf, cache), eidSession_(nullptr) {
   theGraph_ = std::make_unique<HGCGraph>();
   min_cos_theta_ = conf.getParameter<double>("min_cos_theta");
   min_cos_pointing_ = conf.getParameter<double>("min_cos_pointing");
   missing_layers_ = conf.getParameter<int>("missing_layers");
   min_clusters_per_ntuplet_ = conf.getParameter<int>("min_clusters_per_ntuplet");
   max_delta_time_ = conf.getParameter<double>("max_delta_time");
+  eidInputName_ = conf.getParameter<std::string>("eid_input_name");
+  eidOutputNameEnergy_ = conf.getParameter<std::string>("eid_output_name_energy");
+  eidOutputNameId_ = conf.getParameter<std::string>("eid_output_name_id");
+  eidMinClusterEnergy_ = conf.getParameter<double>("eid_min_cluster_energy");
+  eidNLayers_ = conf.getParameter<int>("eid_n_layers");
+  eidNClusters_ = conf.getParameter<int>("eid_n_clusters");
+
+  // mount the tensorflow graph onto the session when set
+  const TrackstersCache *trackstersCache = dynamic_cast<const TrackstersCache *>(cache);
+  if (trackstersCache->eidGraphDef != nullptr) {
+    eidSession_ = tensorflow::createSession(trackstersCache->eidGraphDef);
+  }
```

Comment:
else ???

Reply:
When the `eidGraphDef` in the cache is not set (which happens when no graph path was set) the `eidSession_` should not be set as well and the inference is not performed.

Reply:
how does this distinguish a bug upstream from a desired silent do nothing behavior?
The code should not be allowed to silently skip when it was expected to do something.